data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/rewards/GetBlockRewards.java

```diff
@@ -0,0 +1,75 @@
+import static tech.pegasys.teku.infrastructure.http.RestApiConstants.FINALIZED;
+import static tech.pegasys.teku.infrastructure.http.RestApiConstants.TAG_BEACON;
+import static tech.pegasys.teku.infrastructure.http.RestApiConstants.TAG_REWARDS;
+import static tech.pegasys.teku.infrastructure.json.types.CoreTypes.BOOLEAN_TYPE;
+import static tech.pegasys.teku.infrastructure.json.types.CoreTypes.UINT64_TYPE;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.apache.commons.lang3.NotImplementedException;
+import tech.pegasys.teku.api.migrated.BlockRewardData;
+import tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiEndpoint;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
+import tech.pegasys.teku.spec.datastructures.metadata.ObjectAndMetaData;
+
+public class GetBlockRewards extends RestApiEndpoint {
+  public static final String ROUTE = "/eth/v1/beacon/rewards/blocks/{block_id}";
+
+  private static final SerializableTypeDefinition<BlockRewardData> DATA_TYPE =
+      SerializableTypeDefinition.object(BlockRewardData.class)
+          .withField("proposer_index", UINT64_TYPE, BlockRewardData::getProposerIndex)
+          .withField("total", UINT64_TYPE, BlockRewardData::getTotal)
+          .withField("attestations", UINT64_TYPE, BlockRewardData::getAttestations)
+          .withField("sync_aggregate", UINT64_TYPE, BlockRewardData::getSyncAggregate)
+          .withField("proposer_slashings", UINT64_TYPE, BlockRewardData::getProposerSlashings)
+          .withField("attester_slashings", UINT64_TYPE, BlockRewardData::getAttesterSlashings)
+          .build();
+
+  private static final SerializableTypeDefinition<ObjectAndMetaData<BlockRewardData>>
+      RESPONSE_TYPE =
+          SerializableTypeDefinition.<ObjectAndMetaData<BlockRewardData>>object()
+              .name("GetBlockRewards")
+              .withField(
+                  EXECUTION_OPTIMISTIC, BOOLEAN_TYPE, ObjectAndMetaData::isExecutionOptimistic)
+              .withField(FINALIZED, BOOLEAN_TYPE, ObjectAndMetaData::isFinalized)
+              .withField("data", DATA_TYPE, ObjectAndMetaData::getData)
+              .build();
+
+  public GetBlockRewards() {
+    super(
+        EndpointMetadata.get(ROUTE)
+            .operationId("getBlockRewards")
+            .summary("Get Block Rewards")
+            .description("Retrieve block reward info for a single block.")
+            .tags(TAG_BEACON, TAG_REWARDS)
+            .pathParam(PARAMETER_BLOCK_ID)
+            .response(SC_OK, "Request successful", RESPONSE_TYPE)
+            .withNotFoundResponse()
+            .withInternalErrorResponse()
+            .build());
+  }
+
+  @Override
+  public void handleRequest(RestApiRequest request) throws JsonProcessingException {
+    throw new NotImplementedException();
```

Comment:
try to call this, if anyone calls it i think it'll crash... we'd need to list it as a possible response as far as i recall